src/parser/priest/discipline/modules/features/AtonementApplicatorBreakdown.js

```diff
@@ -0,0 +1,262 @@
+    storeShadowMendsCasts(event){
+        this._castsApplyBuffsMap.set({
+            "event": event,
+            "applicatorId": SPELLS.SHADOW_MEND.id,
+        }, null);
+    }
+    
+    storePowerWordShieldsCasts(event){
+        this._castsApplyBuffsMap.set({
+            "event": event,
+            "applicatorId": SPELLS.POWER_WORD_SHIELD.id,
+        }, null);
+    }
+    
+    storePowerWordRadiancesCastTimestamp(event){
+        this._lastRadianceCastTimestamp = event.timestamp;
+    }
+
+    assignAtonementBuffToApplicator(event){
+        if(event.__fabricated === true){
+            return;
+        }
+
+        if(event.timestamp === this._lastRadianceCastTimestamp){ // Power Word: Radiance
+
+            //Set the wasRefreshed property of the old atonement on the same target to true
+            //so we can stop attributing atonement healing to the old atonement
+            if(event.type === "refreshbuff"){
+                this.setWasRefreshedProperty(event, true);
+            }
+
+            //Putting a custom event object for Radiances since there it's only 1 cast for 5 buffs
+            this._castsApplyBuffsMap.set({
+                "event": {
+                    "timestamp": this._lastRadianceCastTimestamp,
+                },
+                "applicatorId": SPELLS.POWER_WORD_RADIANCE.id,
+            }, 
+            {
+                "applyBuff": event,
+                "atonementEvents": [],
+                "extendedByEvangelism": false,
+                "wasRefreshed": false,
+            });
+        } else { //Shadow Mend and Power Word: Shield
+            if(event.type === "refreshbuff"){
+                this.setWasRefreshedProperty(event, true);
+            }
+
+            //Get the latest cast with the corresponding targetID
+            const playerWithAtonement = event.targetID;
+            const reversedMapKeys = Array.from(this._castsApplyBuffsMap.keys()).slice().reverse();
+            const mostRecentCastApplyBuff = reversedMapKeys.find(cast => cast.event && cast.event.targetID === playerWithAtonement);
+            if (mostRecentCastApplyBuff) {
+                this._castsApplyBuffsMap.set(mostRecentCastApplyBuff, {
+                    "applyBuff": event,
+                    "atonementEvents": [],
+                    "extendedByEvangelism": false,
+                    "wasRefreshed": false,
+                });
+            }
+        }
+    }
+
+    handleAtonementsHits(event){
+        //Healing from atonements pre-applied before entering combat
+        //will assume PW:S as the applicator since it's usually the most common one used pre-pull,
+        const atonementBuffs = this._castsApplyBuffsMap.values();
+        if(Array.from(atonementBuffs).find(atonement => atonement === null || atonement.applyBuff.targetID === event.targetID) === undefined){
+            this._prepullApplicatorHealing += event.amount;
+        }
+
+        for(const [cast, atonement] of this._castsApplyBuffsMap){
```

Comment:
This is entirely personal preference, but you should look into more functional methods of iteration that will allow you to improve composition in the future.

For example,

```js
this._castsApplyBuffsMap.forEach(([cast, atonement]) => {...});
```